components/infobox/wikis/starcraft2/infobox_person_custom_shared.lua

```diff
@@ -84,57 +71,67 @@ function CustomPerson.shouldStoreData()
 end
 
 function CustomPerson.nameDisplay()
-	CustomPerson.getRaceData(_args.race or 'unknown')
-	local raceIcon = RaceIcon({'alt_' .. _raceData.race})
+	local raceData = CustomPerson.readFactions(_args.race or Faction.defaultFaction)
+
+	local raceIcons
+	if raceData.isAll then
+		raceIcons = RACE_ALL_ICON
+	else
+		raceIcons = table.concat(Array.map(raceData.factions, function(faction)
+			return Faction.Icon{faction = faction, size = 'large'}
+		end))
+	end
+
 	local name = _args.id or _PAGENAME
 
-	return raceIcon .. '&nbsp;' .. name
+	return raceIcons .. '&nbsp;' .. name
 end
 
 function CustomPerson.getRaceData(race, asCategory)
-	race = string.lower(race)
-	race = CleanRace[race] or race
-	local raceTable = _RACE_DATA[race]
+	local factions = CustomPerson.readFactions(race).factions
+
+	return table.concat(Array.map(factions, function(faction)
+		if asCategory then
+			return '[[:Category:' .. faction .. ' Players|' .. faction .. ']]'
+		end
+		return '[[' .. faction .. ']]'
+	end) or {}, ',&nbsp;')
+end
 
-	local faction, faction2
-	if race == _RACE_ALL_SHORT then
-		faction = _RACE_ALL
+function CustomPerson.readFactions(input)
+	local factions
+	if input == RACE_ALL or input == RACE_ALL_SHORT then
+		factions = Table.copy(Faction.coreFactions)
 	else
-		faction = (raceTable or {})[1]
-		faction2 = (raceTable or {})[2]
+		factions = Faction.readMultiFaction(input, {alias = false})
 	end
 
-	local display
-	if not raceTable and race ~= 'unknown' then
-		display = '[[Category:InfoboxRaceError]]<strong class="error">' ..
-			mw.text.nowiki('Error: Invalid Race') .. '</strong>'
-	else
-		if asCategory then
-			for raceIndex, raceValue in ipairs(raceTable or {}) do
-				raceTable[raceIndex] = ':Category:' .. raceValue .. ' Players|' .. raceValue .. ']]'
-					.. '[[Category:' .. raceValue .. ' Players'
-			end
-		end
-		if raceTable then
-			display = '[[' .. table.concat(raceTable, ']],&nbsp;[[') .. ']]'
+	local isAll = false
+	-- check if factions == Faction.coreFactions modulo order
+	if #factions == Table.size(Faction.coreFactions) then
```

Comment:
```suggestion
	if #factions == #Faction.coreFactions then
```

Reply:
does not work due to coreFactions containing a meta table

Reply:
got removed anyways due to the other comment

Reply:
Interesting, I didn't know mw.loadData added a metatable that prevents arrays from behaving like arrays. Was that always the case?

Reply:
they behave like arrays in most cases
like you can loop etc but the `#` fails